### ISSUE TITLE
vmcloak-install crashes with adobe9

### ISSUE THREAD

**Issue by Liikt** [jbremer/vmcloak#43]
The error:

```
Traceback (most recent call last):
  File "/usr/local/bin/vmcloak-install", line 5, in module
    pkg_resources.run_script('VMCloak==0.3.10', 'vmcloak-install')
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 528, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1394, in run_script
    execfile(script_filename, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/EGG-INFO/scripts/vmcloak-install", line 105, in module
    main()
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/EGG-INFO/scripts/vmcloak-install", line 92, in main
    d(h, m, a, image, version, settings).run()
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/vmcloak/dependencies/adobe9.py", line 41, in run
    self.wait_process_exit("adobe9.exe")
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/vmcloak/abstract.py", line 339, in wait_process_exit
    for line in self.a.execute("tasklist").json()["stdout"].split("\n"):
KeyError: 'stdout'
```

**Comment by jbremer** [comment]
Oh, forgot to reply here, sorry. Did you already resolve this issue in the meantime?
It would appear there's an issue with the returned json.. :)
Can you reproduce this issue?

**Comment by Liikt** [comment]
This issue always happens when i try to install adobe. I tried it several times and i always had this happen. But no i sadly did not resolve it in the meantime. I think i tried it with office. I am not sure on this tho. I can tell you more tomorrow.

**Comment by Liikt** [comment]
Okey so office produces the same kind of error. After office I tried a different program and i got a different error this time 

```
VBoxManage: error: Machine settings file '<dir>/.vmcloak/vms/foo/foo.vbox' already exists
VBoxManage: error: Details: code VBOX_E_FILE_ERROR (0x80bb0004), component MachineWrap, interface IMachine, callee nsISupports
VBoxManage: error: Context: "CreateMachine(bstrSettingsFile.raw(), bstrName.raw(), ComSafeArrayAsInParam(groups), bstrOsTypeId.raw(), createFlags.raw(), machine.asOutParam())" at line 272 of file VBoxManageMisc.cpp
ERROR:vmcloak.vm:[-] Error running command: Command '['/usr/bin/VBoxManage', 'createvm', '--register', '--name', 'foo', '--basefolder', '<dir>/.vmcloak/vms']' returned non-zero exit status 1
Traceback (most recent call last):
  File "/usr/local/bin/vmcloak-install", line 5, in <module>
    pkg_resources.run_script('VMCloak==0.3.10', 'vmcloak-install')
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 528, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1394, in run_script
    execfile(script_filename, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/EGG-INFO/scripts/vmcloak-install", line 105, in <module>
    main()
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/EGG-INFO/scripts/vmcloak-install", line 51, in main
    m.create_vm()
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/vmcloak/vm.py", line 76, in create_vm
    basefolder=vms_path, register=True)
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/vmcloak/vm.py", line 35, in _call
    raise CommandError
vmcloak.exceptions.CommandError
```
(I replaced parts of the path with <dir> but those are the actual paths)

If you then try to clone the VM with this error  you get yet another different error:

```
VBoxManage: error: Failed to lock source media '<dir>/.vmcloak/image/foo.vdi'
VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component MediumWrap, interface IMedium, callee nsISupports
VBoxManage: error: Context: "CloneTo(pDstMedium, ComSafeArrayAsInParam(l_variants), NULL, pProgress.asOutParam())" at line 893 of file VBoxManageDisk.cpp
ERROR:vmcloak.vm:[-] Error running command: Command '['/usr/bin/VBoxManage', 'clonehd', u'<dir>/.vmcloak/image/foo.vdi', '<dir>/.vmcloak/image/foo1.vdi']' returned non-zero exit status 1
Traceback (most recent call last):
  File "/usr/local/bin/vmcloak-clone", line 5, in <module>
    pkg_resources.run_script('VMCloak==0.3.10', 'vmcloak-clone')
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 528, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1394, in run_script
    execfile(script_filename, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/EGG-INFO/scripts/vmcloak-clone", line 47, in <module>
    main()
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/EGG-INFO/scripts/vmcloak-clone", line 34, in main
    m.clone_hd(image.path, outpath)
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/vmcloak/vm.py", line 120, in clone_hd
    self._call('clonehd', hdd_inpath, hdd_outpath)
  File "/usr/local/lib/python2.7/dist-packages/VMCloak-0.3.10-py2.7.egg/vmcloak/vm.py", line 35, in _call
    raise CommandError
vmcloak.exceptions.CommandError
```

**Comment by Liikt** [comment]
Oh what I forgot to mention is that I try to install the programs on a winxp VM. Maybe that is the cause of the Problem.

**Comment by jbremer** [comment]
So one of the issues is that the VM already exists. So basically VMCloak creates a temporary VM (e.g., `foo.vdi`) which it uses to do whatever you tell it to (in this case, install adobe or office). Given that the VM already exists it can't do its job as VirtualBox will fail when trying to create a VM with a duplicate name.
Please try either deleting that VM (`foo`) or creating a new VMCloak *image* with a new name, etc.

Windows XP and Windows 7 have both been tested plenty of times and should work mostly out of the box.

**Comment by Svieg** [comment]
In my experience, deleting the *.vbox in $HOME/.vmcloak/vms/[VM_NAME]/VM_NAME.vbox works. I did that a lot when my scripts failed and I have yet to see a problem caused by this. The VBoxManage unregistervm basically does that if you decide to keep the files if I am not mistaken. Jurriaan, would you see a potential problem with that ?

**Comment by jbremer** [comment]
Yeah, that should be fine  :)

**Comment by Liikt** [comment]
Well i knew that i just have to delete and then reinstall the VM to get it accessable again. But that sadly doesn't fix the initial issue.

**Comment by razuz** [comment]
is this still an issue ? AFAIK the main lock issues which cause random other issues should be resolved now with 3.12

**Comment by jbremer** [comment]
What @razuz said - please re-open or create a new issue if you run into any other problems.
